cmake/CMakeLists.txt

```diff
@@ -197,6 +203,20 @@ if(onnxruntime_USE_PREBUILT_PB)
     include(${_PROTOBUF_INSTALL_PREFIX}/lib64/cmake/protobuf/protobuf-config.cmake)
   endif()
   include(protobuf_function.cmake)
+
+# mode 2
+elseif(NOT "${ONNX_CUSTOM_PROTOC_EXECUTABLE}" STREQUAL "")
+  add_subdirectory(${PROJECT_SOURCE_DIR}/external/protobuf/cmake EXCLUDE_FROM_ALL)
+  set_target_properties(libprotobuf PROPERTIES FOLDER "External/Protobuf")
```

Comment:
I don't think this change is needed.
Even without that, I can do cross-compiling for ARM, by using a prebuilt protoc. 
Please look at protobuf_function.cmake. That file handles this.

Reply:
This change is needed irrespective to be consistent with the original author's three options to cross this hurdle (listed in the comments above this section).

This entire change is to make it easy to setup a CI build for ARM builds. As you rightly say, it's easy enough for anyone to cross-compile locally using a pre-built protoc as it is. But thischange makes it easier to setup a CI pipeline. What do you think ?

Reply:
I'm the original author :-)

Why do you need to set _PROTOBUF_INSTALL_PREFIX in the mode 2?

Reply:
Got your point. Thanks. Reverting it back to how this section was originally.